src/herder/HerderImpl.cpp

```diff
@@ -189,6 +189,17 @@ HerderImpl::valueExternalized(uint64 slotIndex, StellarValue const& value)
         static_cast<uint32>(slotIndex),
         getSCP().getExternalizingState(slotIndex));
 
+    // reflect upgrades with the ones included in this SCP round
+    {
+        bool updated;
+        auto newUpgrades = mUpgrades.removeUpgrades(
+            value.upgrades.begin(), value.upgrades.end(), updated);
+        if (updated)
```

Comment:
I think it would be nicer to say here `if (newUpgrades != mUpgrades)`. It would require implementing `!=`.

Reply:
yeah if we end up needing to compare we can do that at some point